x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/nvidia/embeddings/NvidiaEmbeddingsServiceSettingsTests.java

```diff
@@ -0,0 +1,551 @@
+        );
+    }
+
+    public void testFromMap_NegativeInputTokens_ThrowsException() {
+        var thrownException = expectThrows(
+            ValidationException.class,
+            () -> NvidiaEmbeddingsServiceSettings.fromMap(
+                buildServiceSettingsMap(
+                    MODEL_VALUE,
+                    CORRECT_URL_VALUE,
+                    SIMILARITY_MEASURE_VALUE.toString(),
+                    DIMENSIONS_VALUE,
+                    MAX_INPUT_TOKENS_NEGATIVE_VALUE,
+                    new HashMap<>(Map.of(RateLimitSettings.REQUESTS_PER_MINUTE_FIELD, RATE_LIMIT_VALUE))
+                ),
+                ConfigurationParseContext.PERSISTENT
+            )
+        );
+        assertThat(
+            thrownException.getMessage(),
+            containsString(
+                Strings.format(
+                    "Validation Failed: 1: [service_settings] Invalid value [%s]. [max_input_tokens] must be a positive integer;",
+                    MAX_INPUT_TOKENS_NEGATIVE_VALUE
+                )
+            )
+        );
+    }
+
+    public void testFromMap_NoRateLimit_Success() {
+        var serviceSettings = NvidiaEmbeddingsServiceSettings.fromMap(
+            buildServiceSettingsMap(
+                MODEL_VALUE,
+                CORRECT_URL_VALUE,
+                SIMILARITY_MEASURE_VALUE.toString(),
+                DIMENSIONS_VALUE,
+                MAX_INPUT_TOKENS_VALUE,
+                null
+            ),
+            ConfigurationParseContext.PERSISTENT
+        );
+
+        assertThat(
+            serviceSettings,
+            is(
+                new NvidiaEmbeddingsServiceSettings(
+                    MODEL_VALUE,
+                    CORRECT_URL_VALUE,
+                    DIMENSIONS_VALUE,
+                    SIMILARITY_MEASURE_VALUE,
+                    MAX_INPUT_TOKENS_VALUE,
+                    new RateLimitSettings(3000)
+                )
+            )
+        );
+    }
+
+    public void testToXContent_WritesAllValues() throws IOException {
+        var entity = new NvidiaEmbeddingsServiceSettings(
+            MODEL_VALUE,
+            CORRECT_URL_VALUE,
+            DIMENSIONS_VALUE,
+            SIMILARITY_MEASURE_VALUE,
+            MAX_INPUT_TOKENS_VALUE,
+            new RateLimitSettings(3)
+        );
+
+        XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
+        entity.toXContent(builder, null);
+        String xContentResult = Strings.toString(builder);
+
+        assertThat(xContentResult, is(XContentHelper.stripWhitespace(Strings.format("""
+            {
+                "model_id": "%s",
+                "url": "%s",
+                "rate_limit": {
+                    "requests_per_minute": 3
+                },
+                "dimensions": 384,
+                "similarity": "dot_product",
+                "max_input_tokens": 128
```

Comment:
Could the values for requests per minute, dimensions, similarity and max input tokens also be placeholders, like the model id and url?